lib/server/encryption.ts

```diff
@@ -0,0 +1,83 @@
+import "server-only";
+import assert from "assert";
+import crypto from "crypto";
+
+const ENCRYPTION_KEY = process.env.ENCRYPTION_KEY!; // Must be 32 bytes (256 bits)
+const ENCRYPTION_IV_LENGTH = 16; // 16 bytes for AES
+
+assert(!!ENCRYPTION_KEY, "ENCRYPTION_KEY environment variable is required");
+assert(Buffer.from(ENCRYPTION_KEY, "hex").length === 32, "ENCRYPTION_KEY must be 32 bytes (64 hex characters)");
+
+export type EncryptedApiKey = string;
+
+export class EncryptionError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "EncryptionError";
+  }
+}
+
+export function encryptApiKey(apiKey: string): EncryptedApiKey {
```

Comment:
There's nothing API key specific about this method right?  Think about changing the signature to something like this:


```suggestion
export function encrypt(plainText: string): string {
```